File: algorithms/ar-slpsrt/golang/sleep_sort.go
package main

import (
	"fmt"
	"time"
)

func main() {
	list := []uint{90, 10, 80, 20, 70, 30, 60, 40, 50}
	fmt.Println("Original list:", list)
	sortedList := sleepSort(list)
	fmt.Println("Sorted list: ", sortedList)
}

func sleepSort(list []uint) []uint {
	sortedList := []uint{}
	out := make(chan uint)
	for _, i := range list {
		go func(n uint) {
			time.Sleep(time.Duration(n) * time.Millisecond)
			out <- n
		}(i)
	}
	for range list {
		sortedList = append(sortedList, <-out)
	}
	return (sortedList)
}

File: algorithms/ar-slpsrt/python2/sleep_sort.py
from time import sleep
from threading import Timer

def sleepSort(values):
    sortedList = []
    def add(x):
        sortedList.append(x)
    max = values[0]
    for value in values:
        if max < value: max = value
        Timer(value, add, [value]).start()
    sleep(max+1)
    return sortedList

list = [3,2,4,5,1]
print "Original list:", list
sortedList = sleepSort(list)
print "Sorted list:", sortedList
